### PR TITLE
[Feature] Created public profile page

### LINKED ISSUE #223: Create Profile Page for ALL Logged in Users
Over time, writers will author multiple Prompts and Entries. The proposal here is that everything the writer posts is available on their own profile page.

In addition, we can have the social networks and a brief biography of the writer.

### PR DESCRIPTION
**Pull request by arnonrdp**
- Created new public profile page for all users.

Resolve #223

### CODE REVIEW

**Review by stravid87 — CHANGES_REQUESTED** [review]
Authors page looks great -- too easy for you I think!
I tried: changing my photo; changing my social media handles; changing my bio -- all works. 
Looks good on mobile and on desktop. It's only June 7th though so let's get the two testing runs to work first and add at least, a testing template for the profile page. I'll fix the currently broken Vitest file. Can you add a user.spec.js? One simple test. Adequate for now I'm thinking.

File: src/pages/PublicProfilePage.vue [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+const socialNetworks = [
+  { name: 'facebook', link: 'https://facebook.com/' },
+  { name: 'instagram', link: 'https://instagram.com/' },
+  { name: 'linkedin', link: 'https://linkedin.com/in/' },
+  { name: 'telegram', link: 'https://web.telegram.org/a/#' },
+  { name: 'twitter', link: 'https://twitter.com/' }
+]
+
+onMounted(async () => {
```
[the remark this answers is not in the document]
Comment (arnonrdp) [reply src/pages/PublicProfilePage.vue:70]: You are right.
I need to improve the query to the user. Let me focus on it today.

File: src/router/routes.js [hunk 2]
```diff
@@ -29,6 +29,10 @@ const routes = [
         path: ':year(\\d{4})/:month(\\d{2})/:id',
         component: () => import('pages/EntryPage.vue')
       },
+      {
+        path: 'fan/:username',
```
[the remark this answers is not in the document]
Comment (arnonrdp) [reply src/router/routes.js:33]: # 🎯

File: src/components/Posts/ThePost.vue [hunk 3]
```diff
@@ -59,7 +59,7 @@
       </section>
       <ShowcaseArt v-if="post?.showcase?.arts?.length" :showcase="post.showcase" />
       <q-separator inset spaced />
-      <section v-if="post?.author" class="flex items-center no-wrap q-pa-md">
+      <section v-if="post?.author" class="flex items-center no-wrap q-pa-md" @click="openProfile">
```
[the remark this answers is not in the document]
Comment (arnonrdp) [reply src/components/Posts/ThePost.vue]: We can do that, but it wouldn't be a profile page.
It would look something like a profile dialog.

**Comment by arnonrdp** [comment]
> @arnonrdp I think that Cypress commenting we can ignore. The test for login-logout is probably worth fixing now.

Alright, by the way feel free to commit update on tests in this branch.

File: src/components/Posts/ThePost.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (stravid87) [reply src/components/Posts/ThePost.vue]: You mean like a tool tip? Something that pop us when the user hovers?

File: src/pages/PublicProfilePage.vue [hunk 4]
```diff
@@ -68,11 +68,7 @@ const socialNetworks = [
 ]
 
 onMounted(async () => {
-  if (!userStore.getUsers.length) {
-    await userStore.fetchUsers()
-  }
-
-  user.value = userStore.getUsers.find((user) => [user.username, user.uid].includes(router.currentRoute.value.params.username))
+  await userStore.getUserByUidOrUsername(router.currentRoute.value.params.username).then((res) => (user.value = res))
```
Comment (stravid87) [inline src/pages/PublicProfilePage.vue:71]: nice.

**Comment by stravid87** [comment]
@arnonrdp 
Please take a look at line 162 of prompts.js specifically. I added an "await" in front of Promise.All(...). Though I doubt this will break the site, it is a change to one of a Pinia store and therefore may have ramifications for testing. 

Before we merge, can you add a single Vitest `<store>.spec` to our repo with just a single test. If you take a look at the vitest files I added, it should be quite straight forward because you can use the other specs as templates.

@hermannleboss, can you figure out what is wrong with our Cypress test on this branch?

File: src/components/Posts/ThePost.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (arnonrdp) [reply src/components/Posts/ThePost.vue]: I meant like a popup (`q-dialog`).

Yes, we can add links to all the places where we have a picture of the user and that way we will redirect to their profile.

We just can't have it in `PublicProfilePage.vue` as that is the destination and not the source.

I will include the links to the profile in the respective sources.

**Comment by hermannleboss** [comment]
> @arnonrdp Please take a look at line 162 of prompts.js specifically. I added an "await" in front of Promise.All(...). Though I doubt this will break the site, it is a change to the Pinia store and therefore may have ramifications outside of testing.

I add a fix, let see if the CI will pass

**Comment by arnonrdp** [comment]
> > @arnonrdp Please take a look at line 162 of prompts.js specifically. I added an "await" in front of Promise.All(...). Though I doubt this will break the site, it is a change to the Pinia store and therefore may have ramifications outside of testing.
> 
> I add a fix, let see if the CI will pass

Thanks for that, Hermann! 💪

**Comment by arnonrdp** [comment]
@stravid87, it is all done.

Can you merge it?

**Comment by stravid87** [comment]
> @stravid87, it is all done.
> 
> Can you approve and merge it?

We've got to get into the habit of building out the test suites as we go -- just add one more spec file. The Public Profile Page Pulls from these four stores. 

- const entryStore = useEntryStore()
- const errorStore = useErrorStore()
- const promptStore = usePromptStore()
- const userStore = useUserStore()

Just add a single test to Vitest or Cypress. It could be as simple as scaffolding the ErrorStore.

**Review by stravid87 — CHANGES_REQUESTED** [review]
We're regressing on our test coverage so, as per our rule, we need to add testing before we get this merged: Vitest or Cypress.

**Comment by arnonrdp** [comment]
> > @stravid87, it is all done.
> > Can you approve and merge it?
> 
> We've got to get into the habit of building out the test suites as we go -- just add one more spec file. The Public Profile Page Pulls from these four stores.
> 
> * const entryStore = useEntryStore()
> * const errorStore = useErrorStore()
> * const promptStore = usePromptStore()
> * const userStore = useUserStore()
> 
> Just add a single test to Vitest or Cypress. It could be as simple as scaffolding the ErrorStore.

Sorry, I missed that. I am on it.

**Review by stravid87 — APPROVED** [review]

**Pull request merged**